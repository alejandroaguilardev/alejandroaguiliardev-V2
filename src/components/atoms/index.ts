export * from "./CardContent/CardContent";
export * from "./Copyright/Copyright";
export * from "./Curriculum/Curriculum";
export * from "./ListOption/ListOption";
export * from "./Logotype/Logotype";
export * from "./Hero/Hero";
export * from "./Networks/Networks";
export * from "./PageRoute/PageRoute";
export * from "./TitleSection/TitleSection";
