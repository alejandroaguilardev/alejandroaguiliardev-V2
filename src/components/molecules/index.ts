export * from './CardItem/CardItem'
export * from './Menu/Menu'
export * from './MenuResponsive/MenuResponsive'
export * from './SocialNetworks/SocialNetworks'