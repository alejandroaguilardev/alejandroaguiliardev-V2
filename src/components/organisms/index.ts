export * from "./About/About";
export * from "./ContainerSocialNetworks/ContainerSocialNetworks";
export * from "./Education/Education";
export * from "./Experience/Experience";
export * from "./Footer/Footer";
export * from "./Navbar/Navbar";
export * from "./Projects/Projects";
export * from "./Skills/Skills";
