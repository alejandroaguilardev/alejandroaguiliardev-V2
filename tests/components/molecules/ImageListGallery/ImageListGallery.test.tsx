import { render, screen } from "@testing-library/react";
import { ImageListGallery } from "../../../../src/components/molecules";

describe("ImageListGallery Test", () => {
	const data = [
		{ img: "/img/banner.jpg", title: "Title 1", author: "author 1" },
		{ img: "/img/banner.jpg", title: "Title 2", author: "author 2" },
	];

	test("toMacthSnapshot", () => {
		const { container } = render(<ImageListGallery dataImage={data} />);

		expect(container).toMatchSnapshot();
		expect(screen.getByText(data[1].title)).toBeTruthy();
		expect(screen.getByText(data[1].author)).toBeTruthy();
	});
});
