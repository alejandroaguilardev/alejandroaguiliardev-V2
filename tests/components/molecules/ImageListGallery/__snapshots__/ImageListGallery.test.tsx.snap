// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageListGallery Test toMacthSnapshot 1`] = `
<div>
  <ul
    class="MuiImageList-root MuiImageList-masonry css-10riir9-MuiImageList-root"
    style="column-count: 3; column-gap: 20px;"
  >
    <li
      class="MuiImageListItem-root MuiImageListItem-masonry css-l086p4-MuiImageListItem-root"
      style="height: auto; margin-bottom: 20px;"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27300%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Title 1"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="MuiImageListItemBar-root MuiImageListItemBar-positionBottom css-ztrqiq-MuiImageListItemBar-root"
      >
        <div
          class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBottom MuiImageListItemBar-titleWrapActionPosRight css-14q7c0c-MuiImageListItemBar-titleWrap"
        >
          <div
            class="MuiImageListItemBar-title css-dasnyc-MuiImageListItemBar-title"
          >
            Title 1
          </div>
          <div
            class="MuiImageListItemBar-subtitle css-hk6gwf-MuiImageListItemBar-subtitle"
          >
            author 1
          </div>
        </div>
        <div
          class="MuiImageListItemBar-actionIcon MuiImageListItemBar-actionIconActionPosRight css-mjbero-MuiImageListItemBar-actionIcon"
        >
          <button
            aria-label="info about Title 1"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1lujp9z-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </li>
    <li
      class="MuiImageListItem-root MuiImageListItem-masonry css-l086p4-MuiImageListItem-root"
      style="height: auto; margin-bottom: 20px;"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27300%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Title 2"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="MuiImageListItemBar-root MuiImageListItemBar-positionBottom css-ztrqiq-MuiImageListItemBar-root"
      >
        <div
          class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBottom MuiImageListItemBar-titleWrapActionPosRight css-14q7c0c-MuiImageListItemBar-titleWrap"
        >
          <div
            class="MuiImageListItemBar-title css-dasnyc-MuiImageListItemBar-title"
          >
            Title 2
          </div>
          <div
            class="MuiImageListItemBar-subtitle css-hk6gwf-MuiImageListItemBar-subtitle"
          >
            author 2
          </div>
        </div>
        <div
          class="MuiImageListItemBar-actionIcon MuiImageListItemBar-actionIconActionPosRight css-mjbero-MuiImageListItemBar-actionIcon"
        >
          <button
            aria-label="info about Title 2"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1lujp9z-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
